.github/workflows/test.yml

```diff
@@ -0,0 +1,89 @@
+name: Quality Assurance
+
+on:
+  push:
+  pull_request:
+
+jobs:
```

Comment:
```suggestion
jobs:
  permissions:
    contents: read
```

Reply:
Put the `permission` block under the `build-linux` block.